osu.Server.Spectator/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using Microsoft.Extensions.DependencyInjection;
+using osu.Game.Online.Rooms;
+using osu.Server.Spectator.Database;
+using osu.Server.Spectator.Entities;
+using osu.Server.Spectator.Hubs;
+
+namespace osu.Server.Spectator.Extensions
+{
+    public static class ServiceCollectionExtensions
+    {
+        public static IServiceCollection AddHubEntities(this IServiceCollection serviceCollection)
+        {
+            return serviceCollection.AddSingleton<EntityStore<SpectatorClientState>>()
+                                    .AddSingleton<EntityStore<MultiplayerClientState>>()
+                                    .AddSingleton<EntityStore<Room>>();
```

Comment:
@peppy this seems to be the wrong room type to cache a store of, but you might beat me on fixing that though.